src/commands/transaction/accountmosaicrestriction.ts

```diff
@@ -0,0 +1,121 @@
+import { command, metadata, option } from 'clime';
+import {
+    Account,
+    AccountRestrictionModification,
+    AccountRestrictionModificationAction,
+    AccountRestrictionTransaction,
+    AccountRestrictionType,
+    Deadline,
+    MosaicId,
+    TransactionHttp,
+    UInt64,
+} from 'nem2-sdk';
+import { OptionsResolver } from '../../options-resolver';
+import { ProfileCommand, ProfileOptions } from '../../profile.command';
+import { ModificationActionValidator } from '../../validators/modificationAction.validator';
+import { RestrictionTypeValidator } from '../../validators/restrictionType.validator';
+
+export class CommandOptions extends ProfileOptions {
+    @option({
+        flag: 'p',
+        description: '(Optional) Select between your profiles, by providing a profile (blank means default profile)',
+    })
+    profile: string;
+
+    @option({
+        flag: 't',
+        description: 'restriction type (allow / block)',
+        validator: new RestrictionTypeValidator(),
+    })
+    restrictionType: string;
+
+    @option({
+        flag: 'a',
+        description: 'Modification action. (1: Add, 0: Remove)',
+        validator: new ModificationActionValidator(),
+    })
+    modificationAction: string;
+
+    @option({
+        flag: 'v',
+        description: 'Address to allow/block',
+    })
+    value: string;
+
+    @option({
+        flag: 'f',
+        description: '(Optional) Maximum fee',
+    })
+    maxfee: number;
+}
+
+@command({
+    description: 'Allow or block incoming transactions containing a given set of mosaics.',
+})
+export default class extends ProfileCommand {
+    constructor() {
+        super();
+    }
+
+    @metadata
+    execute(options: CommandOptions) {
+        if (!['allow', 'block'].includes(options.restrictionType)) {
+            options.restrictionType = OptionsResolver(options,
+                'restrictionType',
+                () => undefined,
+                'Fill in the restriction type (allow / block): ');
+        }
+
+        if (!['1', '0'].includes(options.modificationAction)) {
```

Comment:
Remove this check, validation can be done using validators. The option will not be asked if it's already set by default.